src/context.rs

```diff
@@ -468,6 +398,32 @@ impl<A: Actor> Context<A> {
 
         Ok(fut)
     }
+
+    fn inbox(&self) -> impl FusedStream<Item = Either<BroadcastMessage<A>, AddressMessage<A>>> {
```

Comment:
I wonder if BroadcastMessage and AddressMessage could be merged together, having a third variant for BroadcastMessageEnvelope. Could check if that changes the size (it might due to niches, I'm not sure).

Reply:
Also, if this is going to be public, it might be worth considering wrapping the item in an opaque struct so that the loop can't see inside. However, this might be a good feature (telling between Shutdown and normal messages). On the other hand, it might open up ways for people to somewhat misuse the rest of the lifecycle infrastructure; the goal should be to create a pit of success, and having one and only one way to manage the lifecycle is probably a better way to do that